Dungeons/src/main/java/dungeons/Room.java

```diff
@@ -0,0 +1,30 @@
+package dungeons;
+
+import dungeons.factory.RoomFactory;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Random;
+import static dungeons.utils.Constants.*;
+
+public abstract class Room {
+    protected List<Room> adjacentRooms = new ArrayList<>();
+    protected boolean isVisited = false;
+    protected Random random = new Random();
+
+    public abstract void enter(Player player);
+
+    public void generateAdjacentRooms() {
+        int numAdjacent = random.nextInt(MAX_ADJ_ROOMS + 1) + 1;
+        for (int i = 0; i < numAdjacent; i++) {
+            Room newRoom = RoomFactory.createRoom();
+            if (!adjacentRooms.contains(newRoom)) {
```

Comment:
why would the new room already be in the adjacent rooms?
And if it does, the number of adjacent rooms will be smaller than the number you randomized in line 17

Reply:
I will decrese the number in the line 17 then,